tests/kernel/smp/src/main.c

```diff
@@ -208,7 +214,12 @@ static void thread_entry_fn(void *p1, void *p2, void *p3)
 	tinfo[thread_num].executed  = 1;
 	tinfo[thread_num].cpu_id = curr_cpu();
 
-	while (count++ < 5) {
+	/* We use this thread function with spawn_threads helper which increase delay before each
+	 * new thread is spawned. So we need take into account the amount of threads we are going to
+	 * spawn to wait enough time before exiting the thread (alternatively we could just increase
+	 * wait time for all configurations)
+	 */
+	while (count++ < (5 + arch_num_cpus())) {
```

Comment:
MISRA police are going to have you jailed for that side effect in the expression

Reply:
As I remember we don't enforce them in the test code, don't we?